components/merkledb/examples/manual_migration.rs

```diff
@@ -0,0 +1,145 @@
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Shows how to migrate database data. The migration follows the following scenario:
+//!
+//! 1. We create and fill database with random data according to schema defined in the
+//!   `migration::v1` module with the `create_initial_data` method.
+//! 2. We perform migration from the `v1` schema to the `v2` schema
+//!   with the help of the `create_migration` and `migrate_wallets` methods.
+//!   The method transforms the data in the old schema to conform to the new schema.
+//!   The old data is **not** removed at this stage; rather, it exists alongside
+//!   the migrated data. This is useful in case the migration needs to be reverted for some reason.
+//! 3. We complete the migration by calling `flush_migration`. This moves the migrated data
+//!   to its intended place and removes the old data marked for removal.
+
+use exonum_merkledb::{
+    access::Prefixed,
+    migration::{flush_migration, Migration},
+    Fork, ObjectHash, ReadonlyFork,
+};
+
+use migration::{
+    check_data_after_flush, check_data_after_merge, check_data_before_flush, create_initial_data,
+    v1, v2,
+};
+
+mod migration;
+
+fn create_migration(fork: &Fork) {
+    println!("\nStarted migration");
+
+    let new_data = Migration::new("test", fork);
+    let old_data = Prefixed::new("test", fork.readonly());
+
+    let old_schema = v1::Schema::new(old_data);
+    let mut new_schema = v2::Schema::new(new_data);
+
+    // Move `ticker` and `divisibility` to `config`.
+    let config = v2::Config {
+        ticker: old_schema.ticker.get().unwrap(),
+        divisibility: old_schema.divisibility.get().unwrap_or(0),
+    };
+    new_schema.config.set(config);
+    // Mark these two indexes for removal.
+    new_data.create_tombstone("ticker");
+    new_data.create_tombstone("divisibility");
+
+    // Migrate wallets using schema:
+    // `Wallet::public_key` field will be removed.
+    // `Wallet::history_hash` field will be added.
+    // Wallets and history from username Eve will be removed.
+    let new_data = Migration::new("test", fork);
+    let old_data = Prefixed::new("test", fork.readonly());
+    migrate_wallets(new_data, old_data);
+}
+
+/// Provides migration of wallets with schema.
+fn migrate_wallets(new_data: Migration<&Fork>, old_data: Prefixed<ReadonlyFork>) {
+    let old_schema = v1::Schema::new(old_data);
+    let mut new_schema = v2::Schema::new(new_data);
+
+    // Migrate wallets.
+    for (i, (public_key, wallet)) in old_schema.wallets.iter().enumerate() {
+        if wallet.username == "Eve" {
```

Comment:
Nit: seems that this `if` is duplicated in each example, is it possible to encapsulate it to something function like `migration_step` to focus on actual differences between examples?